### ISSUE TITLE
pylint does not recognize potential undefined variables

### ISSUE THREAD

**Issue by pylint-bot** [PyCQA/pylint#85]
Originally reported by: **Mirko Dziadzka (BitBucket: [mirkodziadzka](http://bitbucket.org/mirkodziadzka), GitHub: @mirkodziadzka?)**

---

The following code should generate a warning, that the variable res may be undefined in  the finally clause

```
#!python
try:
    # throw exception here to make res undefined
    res = 42
finally:
    print res
```

---
- Bitbucket: https://bitbucket.org/logilab/pylint/issue/85

**Comment by dpinol** [comment]
this has been open 6 years. Is it on the roadmap?

thanks

**Comment by Pierre-Sassoulas** [comment]
Here's the test case actualized for python3:
```python
try:
    # throw exception here to make res undefined
    res = 1 / 0
    res = 42
finally:
    print(res)
```

@dpinol this require control-flow which pylint does not have. Apparently no one worked on it yet, I'd be glad to review a merge request that fix this.
